### ISSUE TITLE
Please make the app compatible with V23!

### ISSUE THREAD

**Issue by skyrid3r698** [nextcloud/end_to_end_encryption#273]
**Is your feature request related to a problem? Please describe.**
I cannot safely upgrade to NC 23

**Describe the solution you'd like**
Official compatibility with NC 23

**Comment by JimLoose** [comment]
i can confirm, that it is suddenly broken with NC 23 😳

**Comment by sunjam** [comment]
Is anyone able to manually re-enable it on 23?  If not, can you post the errors or logs are seeing.

**Comment by Joluwe** [comment]
looks like they did not push the latest version from Github: https://apps.nextcloud.com/apps/end_to_end_encryption (1.8.1) ... Github Repo is already 1.10.0 and https://github.com/nextcloud/end_to_end_encryption/blob/master/appinfo/info.xml#L22

**Comment by JimLoose** [comment]
looks like they did not push the latest version from Github: https://apps.nextcloud.com/apps/end_to_end_encryption (1.8.1) ... Github Repo is already 1.10.0 and https://github.com/nextcloud/end_to_end_encryption/blob/master/appinfo/info.xml#L22

**Comment by overhacked** [comment]
Sorry to bug anyone, but this is the only blocker for us to upgrade our installation to NC23. Are there any issues that I can help troubleshoot and resolve so that a release for NC23 can be published to the store?

**Comment by bcutter** [comment]
Déjà vu... https://github.com/nextcloud/end_to_end_encryption/issues/245

And again, on top of the list as always: E2E.
![grafik](https://user-images.githubusercontent.com/13799156/154866066-e7875735-218f-4ac0-b7a9-fea8323a0030.png)

What's missing? New version still not landed in NC app store.

**Comment by Rhinou** [comment]
Was this app abandoned? Just want to know if it makes sense to wait, as it is ~3 months since release of NC 23.x, and E2EE is still not compatible...

**Comment by sunjam** [comment]
I believe there is no timeline on it.  Just a guess

**Comment by bcutter** [comment]
When will the new app version compatible with NC v23 arrive in the NC app store?

**Comment by eibex** [comment]
> Was this app abandoned? Just want to know if it makes sense to wait, as it is ~3 months since release of NC 23.x, and E2EE is still not compatible...

It took 4 months to get the NC22 update pushed to the app store

**Comment by mikeperalta1** [comment]
plz

**Comment by bcutter** [comment]
Just hack the max version string in appinfo. Works for NC v23.

Sad to say but we need to help ourself as nothing is going on here.

**Comment by mikeperalta1** [comment]
> Just hack the max version string in appinfo. Works for NC v23.
> 
> Sad to say but we need to help ourself as nothing is going on here.

You're saying install the latest version, then manually edit the php file to reflect the changes already PR'ed?

Would someone be able to just issue a PR somewhere in the app store repo/area to make this happen as well?

Or is there an easy way to install an app directly from github, rather than going through the store?

**Comment by Rhinou** [comment]
If all that's necessary is changing max version string, then why is it taking so long?

**Comment by guseit** [comment]
Is this project abandoned? E2E is crucial for the use of nextcloud, would render all our systems pretty much useless, if this was abandoned. Rolling back from 23 got us in quite a lot of trouble. Any information on the status?

**Comment by bcutter** [comment]
> If all that's necessary is changing max version string, then why is it taking so long? 

I have no idea - and I'm tired of questioning and waiting for months. E2EE simply does not get the love it needs.
But the version string modification works perfectly (for me).

**Comment by guseit** [comment]
@georgehrke @schiessle @nickvergessen @skjnldsv @MorrisJobke @tobiasKaminsky @Valdnet @comradekingu @rullzer @juliushaertl @come-nc @LukasReschke @lufthans @iMarkus @CarlSchwan @jospoortvliet is this project abandoned? :'(

**Comment by LukasReschke** [comment]
I don’t know, I don’t work on Nextcloud anymore. 🤷‍♂️

**Comment by rullzer** [comment]
> I don’t know, I don’t work on Nextcloud anymore. man_shrugging

Same here

**Comment by guseit** [comment]
@nickvergessen / all (former) devs: sry, no offense, don't wanna be pushy on a project that's FOSS. Using abandoned software is just gonna be a compliance problem, so we have to consider migrating away from Nextcloud or stop using E2E. :(

**Comment by tobiasKaminsky** [comment]
If you need it this urgent, feel free to talk to our sales team for a subscription.

**Comment by MorrisJobke** [comment]
> I don’t know, I don’t work on Nextcloud anymore. 🤷‍♂️

Unfortunately the same for me 🙈

**Comment by guseit** [comment]
@tobiasKaminsky actually though about this. It's not urgent like "need it in a couple of weeks", more like "no answer in 4 month - might be gone for good and should not be used ever again.". A short "we're working on this, should be released within n month." would do the job for me.

**Comment by nickvergessen** [comment]
> nickvergessen / all (former) devs: sry, no offense, don't wanna be pushy on a project that's FOSS. Using abandoned software is just gonna be a compliance problem, so we have to consider migrating away from Nextcloud or stop using E2E. :(

@guseit really appreciate how interested you are in the app and your research to try to get the problem resolved. I see you pinged everyone who commited ever to the app, but did not check what they contributed (I'm only a bot in this app and never commited any content, only CI work and generic work I do in a scripted way for 70+ apps). We have a high workload, so we have to prioritise and these pings cause much distraction and stress. Thanks for understanding!
The app is still active, a person was assigned to the task and it will be resolved once it reaches the top of their priorities.

**Comment by Rhinou** [comment]
@guseit  That's the problem, this uncertainty. If it is gone, it is gone. Just announce it, and we either learn to live without it, or move elsewhere. But for a feature still so proudly presented on NC web-page ("Nextcloud features an enterprise-grade, seamlessly integrated solution for end-to-end encryption") there is no info about its future...

**Comment by jospoortvliet** [comment]
> @guseit That's the problem, this uncertainty. If it is gone, it is gone. Just announce it, and we either learn to live without it, or move elsewhere. But for a feature still so proudly presented on NC web-page ("Nextcloud features an enterprise-grade, seamlessly integrated solution for end-to-end encryption") there is no info about its future...

I'm afraid we are 100% transparent: there has never been a decision to NOT work on the E2E app, or to drop it. So there is nothing to announce. It's been on todo lists to update the app forever. But every week we have to pick the tasks to work on for the week, other things were more urgent, be it customer problems and features they pay for or the thousands of other issues people ask about in the community.

I maintain the website and I'd also prefer the site to be in line with what is the state of things is - but I know as much as you do. So my heuristic is usually "if an app is not available for a release that gets updated, it's time to remove the feature from the website". This app is available for Nextcloud 22 from what I can see in the app store, so I haven't thought about removing it yet.

As Joas said, somebody was assigned the task to work on it, but if they get sick or if other higher priority things come up - it might not happen. In the end, it's like in any other open source project: either somebody volunteers or somebody gets paid to do something. Everyone here in the chat is welcome to step up in either of those ways!

**Comment by CarlSchwan** [comment]
I started adding static analysis to the app and porting away from deprecated stuff: https://github.com/nextcloud/end_to_end_encryption/pull/290 in case someone wants to do a quick review ;)

**Comment by guseit** [comment]
> As Joas said, somebody was assigned the task to work on it, but if they get sick or if other higher priority things come up - it might not happen. In the end, it's like in any other open source project: either somebody volunteers or somebody gets paid to do something. Everyone here in the chat is welcome to step up in either of those ways!

hey @jospoortvliet, thanks for the insight. The fact that E2E is quite a big thing on the webpage besides not being available in the latest release is a bit strange, though. 

Weather E2E is dropped completely from nextcloud or not would have a *huge* impact on my job, as some people I council rely on it - especially those who can't afford an officcial subscription, like small NGOs.

I'm totally fine with "someone" taking care of it "at some point". I'd never expect a quick fix from an community-based project or anything (if necessary, I'd do it on my own). For now I'm just gonna stay with 22 or apply the fix mentioned above.

Thanks everyone! <3

**Comment by sunjam** [comment]
If you have client issues I strongly recommend moving all of your systems
to cryptomator.  As long as the e2e app depends on the server you continue
to have this problem. Just my .02 as a random person, so please do whatever
you want.

**Comment by Liberdev404** [comment]
Is this development for this active again? What happened?

**Comment by CarlSchwan** [comment]
https://github.com/nextcloud-releases/end_to_end_encryption/releases/tag/v1.9.1 is released

https://apps.nextcloud.com/apps/end_to_end_encryption

**Comment by jospoortvliet** [comment]
> If you have client issues I strongly recommend moving all of your systems
> to cryptomator.  As long as the e2e app depends on the server you continue
> to have this problem. Just my .02 as a random person, so please do whatever
> you want.

Hmmmm, well, any piece of code has to be maintained. It's true our server moves fast so that takes a bit extra effort, but if cryptomator would not be maintained you also shouldn't use it...

**Comment by sunjam** [comment]
> but if cryptomator would not be maintained you also shouldn't use it...

[It is fully maintained](https://github.com/cryptomator) according to their github.   Seems fine and their most recent release was last week. :shrug: [You actually wrote an article about them on the Nextcloud blog interviewing the CEO](https://nextcloud.com/blog/qa-tobias-hagemann-talks-about-storing-sensitive-files-in-nextcloud/) in 2017, so I figured it was a pretty great recommendation.

> It's true our server moves fast so that takes a bit extra effort

Glad to see the new v23 release.  It isn't just your server moving fast, but the e2e app requiring server access #8  in order to work.  No doubt it is more technical than that, but I'm just happy that it continues to be worked on.  Super tricky problem to solve. :heart:

**Comment by theCalcaholic** [comment]
@jospoortvliet The difference is: Cryptomator can't be broken by a change to Nextcloud Server, as long as the sync clients aren't broken themselves (even then, it could be used over webdav).
The E2EE-App is regularly broken on basically every major server release, which makes it only viable if using a very conservative upgrade strategy and don't care much about the latest and shinyest features :)

**Comment by mikeperalta1** [comment]
@theCalcaholic I have totally given up on E2EE because it's not worked properly once since I started trying to get it working on my NC instance nearly a year ago. Went through 2 server versions and it always ends up stalling and locking up before it's finished. Granted, my server was a bit slow due to use of minio for primary storage, but still. I've switched to normal server encryption and now it's fine.

Meanwhile, I've been using Cryptomator for years and it's never let me down, so I too will use Cryptomator from now on. The only sad part is the NC desktop app won't be able to "see" the raw files for conflict resolution.
